packages/resource-deployment/scripts/delete-pools-if-needed.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set -eo pipefail
+
+export resourceGroupName
+export batchAccountName
+export keyVault
+export enableSoftDeleteOnKeyVault
+export pools
+
+areVmssOld=false
+recycleVmssIntervalDays=15
+
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -r <resource group> -p <parameter template file path> [-d <flag to force pools to drop>]
+"
+    exit 1
+}
+
+. "${0%/*}/process-utilities.sh"
+
+function checkIfVmssAreOld {
+    areVmssOld=false
+    local hasCreatedDateTags=false
+    
+    local createdDates=$(az vmss list \
+        --query "[?tags.BatchAccountName=='$batchAccountName'].tags.VmssCreatedDate" \
+        -o tsv
+    )
+    
+    echo "VMSS created dates: $createdDates"
+
+    for createdDate in $createdDates; do
+        hasCreatedDateTags=true
+        local recycleDate=$(date -d "$createdDate+$recycleVmssIntervalDays days" "+%Y-%m-%d")
+        local currentDate=$(date "+%Y-%m-%d")
+        
+        if [[ "$currentDate" > "$recycleDate" ]] || [[ "$currentDate" == "$recycleDate" ]]; then
+            echo "Found vmss with older created date $createdDate. 
+                Expected to be not older than $recycleVmssIntervalDays days.
+                Marking all vmss as old
+            "
+            areVmssOld=true
+            break
```

Comment:
do you plan to drop all vmss-s if one has expired?

Reply:
Yes, that's what we were doing before as well